source/ranges.tex

```diff
@@ -9341,6 +9357,550 @@
+a sequence of elements.
+
+\pnum
+\indexlibrarymember{enumerate}{views}%
+The name \tcode{views::enumerate} denotes a range adaptor object.
+Given a subexpression \tcode{E},
+the expression \tcode{views::enumerate(E)} is expression-equivalent to
+\tcode{enumerate_view<views::all_t<decltype((E))>>(E)}.
+\begin{example}
+\begin{codeblock}
+vector<int> vec{ 1, 2, 3 };
+for (auto [index, value] : enumerate(vec) )
+  cout << index << ":" << value << ' ';         // prints: \tcode{0:1 1:2 2:3}
+\end{codeblock}
+\end{example}
+
+\rSec3[range.enumerate.view]{Class template \tcode{enumerate_view}}
+
+\indexlibrarymember{begin}{enumerate_view}%
+\indexlibrarymember{end}{enumerate_view}%
+\indexlibrarymember{size}{enumerate_view}%
+\begin{codeblock}
+namespace std::ranges {
+  template<@\libconcept{view}@ V>
+    requires @\exposconcept{range-with-movable-references}@<V>
+  class enumerate_view : public view_interface<enumerate_view<V>> {
+    V @\exposidnc{base_}@ = V();                                    // \expos
+
+    // \ref{range.enumerate.iterator}, class template \tcode{enumerate_view::\exposid{iterator}}
+    template<bool Const>
+    class @\exposidnc{iterator}@;                                   // \expos
+
+    // \ref{range.enumerate.sentinel}, class template \tcode{enumerate_view::\exposid{sentinel}}
+    template<bool Const>
+    class @\exposidnc{sentinel}@;                                   // \expos
+
+  public:
+    constexpr enumerate_view() requires @\libconcept{default_initializable}@<V> = default;
+    constexpr explicit enumerate_view(V base);
+
+    constexpr auto begin() requires (!@\exposconcept{simple-view}@<V>)
+      { return @\exposid{iterator}@<false>(ranges::begin(@\exposid{base_}@), 0); }
+    constexpr auto begin() const requires @\exposconcept{range-with-movable-references}@<const V>
+      { return @\exposid{iterator}@<true>(ranges::begin(@\exposid{base_}@), 0); }
+
+    constexpr auto end() requires (!@\exposconcept{simple-view}@<V>) {
+      if constexpr (@\libconcept{common_range}@<V> && @\libconcept{sized_range}@<V>)
+        return @\exposid{iterator}@<false>(ranges::end(@\exposid{base_}@), ranges::distance(@\exposid{base_}@));
+      else
+        return @\exposid{sentinel}@<false>(ranges::end(@\exposid{base_}@));
+    }
+    constexpr auto end() const requires @\exposconcept{range-with-movable-references}@<const V> {
+      if constexpr (@\libconcept{common_range}@<const V> && @\libconcept{sized_range}@<const V>)
+        return @\exposid{iterator}@<true>(ranges::end(@\exposid{base_}@), ranges::distance(@\exposid{base_}@));
+      else
+        return @\exposid{sentinel}@<true>(ranges::end(@\exposid{base_}@));
+    }
+
+    constexpr auto size()
+      requires @\libconcept{sized_range}@<V>
+      { return ranges::size(@\exposid{base_}@); }
+    constexpr auto size() const
+      requires @\libconcept{sized_range}@<const V>
+      { return ranges::size(@\exposid{base_}@); }
```

Comment:
Wrong indentation. Please, choose the last one (just like `base` below), or any other of these:
![1677013600](https://user-images.githubusercontent.com/21071787/220458796-6b5b833f-2416-4787-bf90-82bc316d766d.png)
![1677013646](https://user-images.githubusercontent.com/21071787/220458799-e83a1975-c13b-4b9b-87a2-56507133f476.png)

Reply:
Thanks!